coreneuron/utils/progressbar/progressbar.c

```diff
@@ -32,6 +32,9 @@ enum { WHITESPACE_LENGTH = 2 };
 /// The amount of width taken up by the border of the bar component.
 enum { BAR_BORDER_WIDTH = 2 };
 
+/// Progress Bar max number of updates
+enum { BAR_UPDATES_NUMBER = 1000 };
```

Comment:
I find this number too large to be a constant. (I don't know if you already agreed with someone else), but 1000 prints is a lot of output.
To my mind we could instead print after N seconds have elapsed

Reply:
This particular number was chosen after a small discussion with @pramodk .
I see your point and I will try to improve this part as you suggested.
Thank you very much! :+1:

Reply:
Another idea. Updates in the first minutes is important so that the user gets to know if something is wrong. 
If the simulation lasts for three days, even with BAR_UPDATES_MAX=1000 he only knows anything after 4.5min. 
@pramodk I don't know if this is overcomplicating, how cool we would like this. Maybe the first X minutes we force updates at rate BAR_UPDATE_INTERVAL, and only after that, we might reduce the rate based on BAR_UPDATES_MAX?

Reply:
I have made a change in the print mechanism based on @ferdonline comments.
Please see the next commit.